packages/roosterjs-editor-dom/lib/utils/setColor.ts

```diff
@@ -36,3 +45,49 @@ export default function setColor(
         }
     }
 }
+
+/**
+ * Change the font color to white or some other color, so the text can be visible with a darker background
+ * @param element The element that contains text.
+ * @param shouldAdaptTheFontColor if true it adapts the font color
+ */
+function adaptFontColorToBackgroundColor(element: HTMLElement, shouldAdaptTheFontColor?: boolean) {
+    if (element.firstElementChild?.hasAttribute('style') || !shouldAdaptTheFontColor) {
+        return;
+    }
+    const backgroundColor = element.style.getPropertyValue('background-color');
+    if (!backgroundColor) {
+        return;
+    }
+    if (isADarkOrBrightColor(backgroundColor) === true) {
+        element.style.color = WHITE;
+    } else if (isADarkOrBrightColor(backgroundColor) === false) {
+        element.style.color = BLACK;
+    }
+}
+
+function isADarkOrBrightColor(color: string): boolean | null {
+    let lightness = calculateLightness(color);
+    if (lightness < DARK_COLORS_LIGHTNESS) {
+        return true;
+    } else if (lightness > BRIGHT_COLORS_LIGHTNESS) {
+        return false;
+    }
+    return null;
+}
+
+/**
+ * Calculate the lightness of HSL (hue, saturation and lightness) representation
+ * @param color a RBG COLOR
+ * @returns
+ */
+function calculateLightness(color: string) {
+    let [r, g, b] = color.match(/[\d\.]+/g) as RegExpMatchArray;
```

Comment:
the Regex should also include "rgb" and "rgba", for example:

/^rgba?\(([\d\.]+)\s*,\s*([\d\.]+)\s*,\s*([\d\.]+)(\s*,\s*[\d\.]+)?\)$/